applications/GeoMechanicsApplication/custom_constitutive/linear_elastic_plane_strain_2D_law.cpp

```diff
@@ -89,12 +101,84 @@ void GeoLinearElasticPlaneStrain2DLaw::CalculatePK2Stress(const Vector& rStrainV
 {
     KRATOS_TRY
 
+    UpdateInternalDeltaStrainVector(rValues);
+
     Matrix C;
     this->CalculateElasticMatrix(C, rValues);
-    // Total formulation
-    noalias(rStressVector) = prod(C, rStrainVector);
+
+    // Incremental formulation
+    noalias(mStressVector) = mStressVectorFinalized + prod(C, mDeltaStrainVector);
+
+    SetExternalStressVector(rStressVector);
 
     KRATOS_CATCH("")
 }
 
+void GeoLinearElasticPlaneStrain2DLaw::UpdateInternalDeltaStrainVector(ConstitutiveLaw::Parameters &rValues){
+    const Vector& rStrainVector = rValues.GetStrainVector();
+    std::transform(rStrainVector.begin(), rStrainVector.begin() + mDeltaStrainVector.size(), mStrainVectorFinalized.begin(), mDeltaStrainVector.begin(), std::minus<>{} );
+}
+
+void GeoLinearElasticPlaneStrain2DLaw::UpdateInternalStrainVectorFinalized(ConstitutiveLaw::Parameters &rValues) {
+    this->SetInternalStrainVector(rValues.GetStrainVector());
+}
+
+void GeoLinearElasticPlaneStrain2DLaw::SetExternalStressVector(Vector& rStressVector){
+    std::copy(mStressVector.begin(), mStressVector.begin() + rStressVector.size(), rStressVector.begin());
+}
+
+void GeoLinearElasticPlaneStrain2DLaw::SetInternalStressVector(const Vector& rStressVector){
+    std::copy(rStressVector.begin(), rStressVector.begin() + mStressVectorFinalized.size(), mStressVectorFinalized.begin());
```

Comment:
In these functions, I'm a bit confused why we use begin iterators of one and the size of another vector. Could they be different sizes, or is there a reason we would like to only copy e.g. a part of rStressVector here?

Reply:
This reflects the setup of the original loop.